sdk/python/arvados/keep.py

```diff
@@ -22,6 +22,16 @@
 global_client_object = None
 
 
+# Monkey patch TCP constants when not available (apple). Values sourced from:
+# http://www.opensource.apple.com/source/xnu/xnu-2422.115.4/bsd/netinet/tcp.h
+if not hasattr(socket, 'TCP_KEEPALIVE'):
+    socket.TCP_KEEPALIVE = 0x010
+if not hasattr(socket, 'TCP_KEEPINTVL'):
+    socket.TCP_KEEPINTVL = 0x101
+if not hasattr(socket, 'TCP_KEEPCNT'):
+    socket.TCP_KEEPCNT = 0x102
```

Comment:
I can add that, but keep will still fail to open socket connections regardless of platform if these constants are not defined. I think this would fail on Windows as well.